src/main/scala/chiseltest/internal/GenericBackend.scala

```diff
@@ -1,217 +1,47 @@
-// SPDX-License-Identifier: Apache-2.0
-
 package chiseltest.internal
-
-import chiseltest.{ClockResolutionException, Region, TimeoutException}
-import chisel3._
-import chiseltest.coverage.{Coverage, TestCoverage}
+import chisel3.{Clock, Data, Module}
 import chiseltest.simulator.SimulatorContext
-import firrtl.AnnotationSeq
-
-import scala.collection.mutable
+import logger.LazyLogging
 
-/** Chiseltest threaded backend using the generic SimulatorContext abstraction from [[chiseltest.simulator]] */
-class GenericBackend[T <: Module](
-  val dut:                T,
-  val dataNames:          Map[Data, String],
-  val combinationalPaths: Map[Data, Set[Data]],
-  tester:                 SimulatorContext,
-  coverageAnnotations:    AnnotationSeq)
-    extends BackendInstance[T]
-    with ThreadedBackend[T] {
+abstract class GenericBackend[T <: Module] extends BackendInterface with LazyLogging {
```

Comment:
A minimal GenericBackend only contains 
1. basic API(peek poke step(I still not so sure step is basic)) implementation
2. dataNames to map Data -> String(I personally this should be a full hierarchy path in the future.)